src/Analyzers/CSharp/CodeFixes/UseImplicitOrExplicitType/UseImplicitTypeCodeFixProvider.cs

```diff
@@ -61,12 +67,12 @@ internal static void ReplaceTypeWithVar(SyntaxEditor editor, SyntaxNode node)
             editor.ReplaceNode(node, implicitType);
         }
 
-        private class MyCodeAction : CodeAction.DocumentChangeAction
+        private class MyCodeAction : CustomCodeActions.DocumentChangeAction
         {
             public MyCodeAction(Func<CancellationToken, Task<Document>> createChangedDocument)
-                : base(CSharpFeaturesResources.use_var_instead_of_explicit_type,
+                : base(Resources.use_var_instead_of_explicit_type,
                        createChangedDocument,
-                       CSharpFeaturesResources.use_var_instead_of_explicit_type)
+                       Resources.use_var_instead_of_explicit_type)
             {
             }
         }
```

Comment:
it feels like all the feaure changes could be pulled out.

Reply:
Changes to actual features, i.e. analyzers/fixers/tests are much lesser compared to other changes in this PR. They can potentially be pulled out into a separate PR, but having them in this PR gives us end to end validation that remainder of changes did not break something in the CodeStyle layer.

Reply:
>  but having them in this PR gives us end to end validation that remainder of changes did not break something in the CodeStyle layer.

this is possible with an 'all up' PR.  i.e. we have the individual PR (X1, x2, x3, etc.) and we have the merged PR (Y).  We gate the individual PRs going on both it and the merged PR passing.  As each individual PR goes in, the merged PR gets smaller and smaller.  Eventually, it's the final PR, with minimal changes in it. :)

Reply:
I believe that's an unnecessary overhead. It's really not that hard to review this PR as a whole.

---
In reply to: [376652211](https://github.com/dotnet/roslyn/pull/41363#discussion_r376652211) [](ancestors = 376652211)

Reply:
> It's really not that hard to review this PR as a whole.

it's literally crushing my browser into a pulp.  even tryin to scroll at times just hangs teh whole browser, then ends up vastly overshooting things.  It's also a PITA because there are indepedendent changes interspersed with each other.  so i have 5 actual different notepads open keeping notes, and i have to keep switching between them to stitch together the whole picture as to what's going with each piece.  There are still parts of this review that i haven't even figured out the purpose of just yet because i haven't seen the whole picture. 

it's really not pleasant.  And having multiple smaller PRs that cover these independent changes is honestly not that hard to do, and is a desirable way to break things up to keep everything manageable and self contained.

I'd also much rather run into a problem in the future and have it trace back to a small and easier to understand PR than a massive one like this that i'd have to page everything back into my head for :-/

Reply:
Having talked to bunch of folks about this, I think a hybrid approach might be better:

1. Pull out pieces that need some semantic changes (API signature changes, breaking type hierarchy, etc.) into small PRs where possible. @CyrusNajmabadi created few of these.
2. @sharwell pulled out a separate PR that just adds empty shared projects: https://github.com/dotnet/roslyn/pull/41550

Let's try to pull out pieces where it seems it will be a real review aid, but not attempt to do so for every piece of functionality that can be pulled apart. We will eventually still have a big PR to review for anything related to moving files or splitting extension methods across files, but that should hopefully be fine to not validate each and every moved extension method.

Reply:
> Let's try to pull out pieces where it seems it will be a real review aid, but not attempt to do so for every piece of functionality that can be pulled apart.

👍  on this.  I'm not asking for every tiny piece to be pulled out.  Just some deteasing of some of hte major chunks to just keep thins more manageable. 

> We will eventually still have a big PR to review for anything related to moving files or splitting extension methods across files, but that should hopefully be fine to not validate each and every moved extension method.

Agreed.  My PR, for example *only* attempts to make such a change, thus making it much easier to review (because there aren't logic changes along with the moves).  I think those tyeps of PRs are much much easier to review, and much faster to get in :)

Thanks for the help here :)

Reply:
@CyrusNajmabadi It's a shame that GitHub PR UX is so bad. Have you tried fetching the PR branch locally and review in VS? It's much easier to see the changes that way.